Gems/PhysX/Code/Source/EditorRigidBodyComponent.cpp

```diff
@@ -216,7 +216,19 @@ namespace PhysX
                         "Include non-simulated shapes in Mass", "If set, non-simulated shapes will also be included in the center of mass, inertia and mass calculations.")
                         ->Attribute(AZ::Edit::Attributes::Visibility, &AzPhysics::RigidBodyConfiguration::GetInertiaSettingsVisibility)
                         ->Attribute(AZ::Edit::Attributes::ChangeNotify, AZ::Edit::PropertyRefreshLevels::EntireTree)
-                    ;
+
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &RigidBodyConfiguration::m_linearAxisLock,
+                        "Linear Axis Lock", "Lock movement along a specified axis (0|1)")
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 1.f)
+                        ->Attribute(AZ::Edit::Attributes::Step, 1.f)
+
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &RigidBodyConfiguration::m_angularAxisLock,
+                        "Angular Axis Lock", "Lock rotation around a specified axis (0|1)")
+                        ->Attribute(AZ::Edit::Attributes::Min, 0.f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 1.f)
+                        ->Attribute(AZ::Edit::Attributes::Step, 1.f)
+                        ;
```

Comment:
What does the UX look like for this?

Is there a reason we didn't make a AxisLock struct that reflects 3 bools to edit contexts. then here have 1 of these structs reflected for each lock here?

Reply:
I was working on the same thing and implemented it with a list of bool flags in a collapsible group. Granted, not the best UX (but I was after a quick fix). 
![image](https://user-images.githubusercontent.com/81370835/127154240-15584b9d-1b38-41c6-9828-0cb4c2e1cb94.png)

Reply:
![image](https://user-images.githubusercontent.com/60428010/127156474-f9d610aa-0e7f-43da-a790-3df7780a78c5.png)
here the picture. 
Later when we get VectorBool3 working, the numbers would be replaced with checkboxes.
I think there's another PR making it a list of checkboxes, so see if that one is preferable

Reply:
the list of check boxes i think is a better UX.
if i had to choose between the vector of floats (that are not really floats) and checkboxes.